doc/source/contributing_docstring.rst

```diff
@@ -0,0 +1,634 @@
+
+            Parameters
+            ----------
+            n : int
+                Number of values to return.
+
+            Return
+            ------
+            pandas.Series
+                Subset of the original series with the n first values.
+
+            See Also
+            --------
+            tail : Return the last n elements of the Series.
+
+            Examples
+            --------
+            >>> import pandas
+            >>> s = pandas.Series(['Ant', 'Bear', 'Cow', 'Dog', 'Falcon',
+            ...                    'Lion', 'Monkey', 'Rabbit', 'Zebra'])
+            >>> s.head()
+            0   Ant
+            1   Bear
+            2   Cow
+            3   Dog
+            4   Falcon
+            dtype: object
+
+            With the `n` parameter, we can change the number of returned rows:
+
+            >>> s.head(n=3)
+            0   Ant
+            1   Bear
+            2   Cow
+            dtype: object
+            """
+            return self.iloc[:n]
+
+Conventions for the examples
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. note::
+   numpydoc recommends avoiding "obvious" imports and importing them with
+   aliases, so for example `import numpy as np`. While this is now an standard
+   in the data ecosystem of Python, it doesn't seem a good practise, for the
+   next reasons:
+
+   * The code is not executable anymore (as doctests for example)
+
+   * New users not familiar with the convention can't simply copy and run it
+
+   * Users may use aliases (even if it is a bad Python practise except
+     in rare cases), but if maintainers want to use `pd` instead of `pandas`,
+     why do not name the module `pd` directly?
+
+   * As this is becoming more standard, there are an increasing number of
+     aliases in scientific Python code, including `np`, `pd`, `plt`, `sp`,
+     `pm`... which makes reading code harder
+
+All examples must start with the required imports, one per line (as
+recommended in `PEP-8 <https://www.python.org/dev/peps/pep-0008/#imports>`_)
+and avoiding aliases. Avoid excessive imports, but if needed, imports from
+the standard library go first, followed by third-party libraries (like
+numpy) and importing pandas in the last place.
+
+When illustrating examples with a single `Series` use the name `s`, and if
+illustrating with a single `DataFrame` use the name `df`. If a set of
+homogeneous `Series` or `DataFrame` is used, name them `s1`, `s2`, `s3`...
+or `df1`, `df2`, `df3`... If the data is not homogeneous, and more than
+one structure is needed, name them with something meaningful, for example
+`df_main` and `df_to_join`.
+
+Data used in the example should be as compact as possible. The number of rows
+is recommended to be 4, unless the example requires a larger number. As for
+example in the head method, where it requires to be higher than 5, to show
+the example with the default values.
+
+Avoid using data without interpretation, like a matrix of random numbers
```

Comment:
There's an issue about using common, meaningful datasets for these. Maybe we can make a decision on that before the sprint (will try to find link later).

Reply:
I also remind such a discussion, but didn't find an open issue, only a discussion in a PR (with mention of gitter), so opened a new issue: https://github.com/pandas-dev/pandas/issues/19710